File: backend/app/core/schema/Accounts/Request.py
from pydantic import BaseModel


class CreateRequest(BaseModel):
    user_id: str
    bank_name: str
    account_number: str | None
    account_type: str
    balance: float | None


class UpdateRequest(BaseModel):
    id: str
    user_id: str
    bank_name: str
    account_number: str | None
    account_type: str
    balance: float | None


class UpdateBalanceRequest(BaseModel):
    id: str
    user_id: str
    balance: float


class DeleteRequest(BaseModel):
    id: str
    user_id: str

File: findrey-server/server/core/dto/account_dto/request.go
package account_dto

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type AccountRequestDTO struct {
	ID primitive.ObjectID `json:"id" binding:"required"`
}

type CreateAccountRequestDTO struct {
	UserID        primitive.ObjectID `json:"userId" binding:"required"`
	BankName      string             `json:"bankName"`
	AccountNumber string             `json:"accountNumber"`
	AccountName   string             `json:"accountName"`
	AccountType   primitive.ObjectID `json:"accountType"`
	Balance       float64            `json:"balance"`
}

type UpdateAccountRequestDTO struct {
	ID            primitive.ObjectID  `json:"id" binding:"required"`
	BankName      *string             `json:"bankName"`
	AccountNumber *string             `json:"accountNumber"`
	AccountName   *string             `json:"accountName"`
	AccountType   *primitive.ObjectID `json:"accountType"`
	Balance       *float64            `json:"balance,omitempty"`
}
